Typos: Run 'typos -w' command over docs/

WalterSmuts opened the pull request:
# Description
```
commit 38b854be04b95e01005c1eb2185ec23307d337d9
Author: Walter Smuts <smuts.walter@gmail.com>
Date:   Sun May 14 13:06:29 2023 +0200

    Typos: Run 'typos -w' command over docs/

    Only running over "docs/" directory since it will likely have the
    largest postivie impact and cause fewest issues.

    Typos will do simple find-and-replace when it detects a word not in it's
    dictionary. It does not have any regard for formatting based on
    surrounding context. Care must be taken not no merge variable names in
    same scope etc.

    Typos can be driven by Github Actions:
    https://github.com/crate-ci/typos/blob/master/docs/github-action.md

    Tool: https://github.com/crate-ci/typos

```
and
```
commit 0c29a13c02a0186acd7c1078586d576cb86794c7
Author: Walter Smuts <smuts.walter@gmail.com>
Date:   Sun May 14 13:21:22 2023 +0200

    Typos: specify -> specificy

    Audit of "docs/" typos that did not have single match.
```

Z-snails commented on docs/source/reference/debugging.rst:
```diff
@@ -98,7 +98,7 @@ Specifically dumped are:
    Detaggable argument types,
    Specialised arguments,
-   Inferrable arguments,
+   Inferable arguments,
```
Inferrable (with 2 'r's) is also correct, and the compiler uses it, so could you change this (and the other uses) back to inferrable?

WalterSmuts commented:
Updated and rebased :+1: 

I've also added a commit to change all uses of `Inferable` to `Inferrable` for consistency. But since both are considered correct feel free to drop the last commit.

andrevidela commented:
LGTM
